src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java

```diff
@@ -0,0 +1,199 @@
+                  ));
+
+  private static final List<Condition> CONDITIONS = ImmutableList.of(
+          new Condition("ios_en", "device.os == 'ios' && device.country in ['us', 'uk']")
+                  .setTagColor(TagColor.INDIGO),
+          new Condition("android_en",
+                  "device.os == 'android' && device.country in ['us', 'uk']"));
+
+  private static final Version VERSION = Version
+          .withDescription(String.format("promo config %s", timestamp));
+
+  @BeforeClass
+  public static void setUpClass() {
+    remoteConfig = FirebaseRemoteConfig.getInstance(IntegrationTestUtils.ensureDefaultApp());
+  }
+
+  @Test
+  public void testTemplateOperations() throws FirebaseRemoteConfigException {
+    // get template to fetch the active template with correct etag
+    final Template oldTemplate = remoteConfig.getTemplate();
+    final Template inputTemplate = Template.fromJSON(oldTemplate.toJSON());
+    final String versionNumber = oldTemplate.getVersion().getVersionNumber();
+
+    // modify template
+    inputTemplate.setParameters(PARAMETERS)
+            .setParameterGroups(PARAMETER_GROUPS)
+            .setConditions(CONDITIONS)
+            .setVersion(VERSION);
+
+    // validate template
+    Template validatedTemplate = remoteConfig.validateTemplate(inputTemplate);
+    assertEquals(inputTemplate.getETag(), validatedTemplate.getETag());
+    assertEquals(PARAMETERS, validatedTemplate.getParameters());
+    assertEquals(PARAMETER_GROUPS, validatedTemplate.getParameterGroups());
+    assertEquals(CONDITIONS, validatedTemplate.getConditions());
+    assertEquals(VERSION, validatedTemplate.getVersion());
+
+    // publish template
+    Template publishedTemplate = remoteConfig.publishTemplate(inputTemplate);
+    assertNotEquals(inputTemplate.getETag(), publishedTemplate.getETag());
+    assertEquals(PARAMETERS, publishedTemplate.getParameters());
+    assertEquals(PARAMETER_GROUPS, publishedTemplate.getParameterGroups());
+    assertEquals(CONDITIONS, publishedTemplate.getConditions());
+    assertNotEquals(VERSION, publishedTemplate.getVersion());
+
+    // get template
+    Template currentTemplate = remoteConfig.getTemplate();
+    assertEquals(publishedTemplate, currentTemplate);
+
+    // get template at version
+    Template atVersionTemplate = remoteConfig.getTemplateAtVersion(versionNumber);
+    assertEquals(oldTemplate, atVersionTemplate);
+    assertEquals(versionNumber, atVersionTemplate.getVersion().getVersionNumber());
+
+    // rollback template
+    Template rolledBackTemplate = remoteConfig.rollback(versionNumber);
+    assertEquals(String.format("Rollback to version %s", versionNumber),
+            rolledBackTemplate.getVersion().getDescription());
+
+    // get template to verify rollback
+    Template activeTemplate = remoteConfig.getTemplate();
+    assertEquals(rolledBackTemplate, activeTemplate);
+  }
+
+  @Test
+  public void testListVersions() throws Exception {
```

Comment:
I'd suggest using the sync APIs here like the previous test case.

Reply:
Updated to use the sync api